### ISSUE TITLE
Flagging Unlikely Departures

### ISSUE THREAD

**Issue by ali206** [206Tours/DB#183]
Hi Brad - Please add this to Phase #7 before moving on to Phase #8.

Can you please implement a way for us to flag certain bookings under certain tours that we don't think will be departing (clients aren't aware yet though)?  By flag i mean provide us with a way to mark each booking under a certain tour (that we don't think will be going) unable to process any further CC payments.  This way, we're not paying the cc fees on clients final payments for a tour that we're inevitably going to cancel and refund back the clients their money.  Implementing this would save us a lot of money on unnecessary cc fees.  

Does that make sense?

If you need further explanation let me know. 

Ex:
Tour SRL040615
Tour MLM041315

**Comment by bradatkins** [comment]
It makes complete sense.

Phase 7 is closed and I'm currently working on Phase 8. I'll mark it as a
future request.

**Comment by ali206** [comment]
Ok i didn' realize phase 7 was closed completely.  

I know phase #8 is only for the website but I don't think this should be postponed so i'm waiting to hear from Milanka if this can be added to phase #8 maybe. 

There's no quick fix to this just for the time being until you have more time to work on it in the future?

These are Chris's suggestions - but i don't know how you would do this without affecting all cc's that need to be processed.  It might only be able to work once we tell you which tours we don't think will be departing.

He could just add a line of code that places a few wrong letters in the CC processing link and makes it unable to process.
It would just be forcing an error when a button is selected. This error would need to be undone when the button is unselected.

**Comment by bradatkins** [comment]
Sorry there is no quick fix.

I don't control the scope of Phase 8. If Milanka wants stuff added, she can
of course add things in there, but previously she wanted to keep the phase
small.

Once you hear back let me know.

**Comment by bradatkins** [comment]
What we can do for this one is apply for an AUTH instead of a SALE for these types of orders. Not sure if you will still be charged for that or not. Can you check with Elavon?

**Comment by ali206** [comment]
This is what i was saying to add to the list that i couldn't quite remember anything about yesterday lol 

I'll check with Elavon but wouldn't it be easier if people registered on a specific tour that we flagged kept getting an error message or something along those lines, so as to not let them even put in their card information? I feel like this shouldn't even involve Elavon.

**Comment by bradatkins** [comment]
We can do whatever you want of course. But if it's flag as unlikely, why
would I even register :)

And if we 'scare' everyone away, we've basically put the kiss of death on
that tour, no?

**Comment by chris206** [comment]
Good point - maybe we could still allow them to enter their payment info, but we would need to hit the button before that info was processed.

Just add a passive layer in between: We dont get charged, they dont notice any difference initially

**Comment by bradatkins** [comment]
Won't work because we don't store the credit card information.

**Comment by ali206** [comment]
I don't understand what this has to do with storing cc info?

We don't need to have their CC stored - just need to not let them process a final payment on their own - or don't allow someone in office to process it for them if they don't know the tour isn't going

**Comment by bradatkins** [comment]
So what if I type in my CC info? Or do we not allow them to pay? I don't understand

**Comment by ali206** [comment]
If a dept is flagged as unlikely to depart, we do not want to allow clients to pay via Credit card. 

I believe Chris' suggestion was to let the client type in their CC info but have the DB always say there's an error so their payment won't actually be processed. Then they'd probably call us and what not. 

I can't think of a better way to implement this but i do think this would create a bit of frustration on clients side and a lot of phone calls to the office.

**Comment by bradatkins** [comment]
Agreed, I don't think that's a sustainable idea....

**Comment by chris206** [comment]
We were talking about this before - apparently office standards have altered so that we are now deciding to cancel departure within 60 days. We do this to save money - but the CC processing fees are staggering.

We have 2 solutions:

1. If a departure is flagged as "unlikely", then the ability to pay with credit card is removed from the options for that order on the client side.

2. Any orders associated to a departure that is "unlikely" will present with some type of alarm bell for the USER at the "orders" level so that USER understands not to charge the account. The ability to pay by check or ACH will go unimpeded.

**Comment by bradatkins** [comment]
This is possible...
